### DISCUSSION TITLE
Azure gpt integration question

### DISCUSSION THREAD

**Discussion by eggcaker** [zed-industries/zed#9044] [Feature Requests]
Some where in the PRs ,i saw the azure integration for assistant, is it support azure gpt for now ? if yes, how to setup them ?

**Comment by Moshyfawn** [comment]
I assume you're referring to #8646. If so, I believe it's already in Zed Preview, as the provider json `type` field already accepts the type `azure_openai`.

You can try to set it up using the sample configuration provided in the above PR.

```json
{
  "assistant": {
    "provider": {
      "type": "azure_openai",
      "api_url": "https://{your-resource-name}.openai.azure.com",
      "deployment_id": "gpt-4",
      "api_version": "2023-05-15"
    }
  },
}
```

**Comment by eggcaker** [comment]
did i compile zed with cargo? just tried 0.125.3 , added above configuration in settings, and setup the key with assitant panel. but it still connect to openai not azure one

**Reply by Moshyfawn** [reply]
You can download the latest Zed Preview build, which is [`0.126.0`](https://zed.dev/releases/preview/0.126.0) as of right now.

**Reply by eggcaker** [reply]
this version works! Thanks!

**Comment by tgds** [comment]
yeah, well, this was removed in #7367, but not sure why.
